python/ray/serve/schema.py

```diff
@@ -978,6 +978,92 @@ class ReplicaDetails(ServeActorDetails, frozen=True):
     )
 
 
+@PublicAPI(stability="alpha")
+class MetricsHealth(str, Enum):
+    HEALTHY = "healthy"
+    DELAYED = "delayed"
+    UNAVAILABLE = "unavailable"
+
+
+@PublicAPI(stability="alpha")
+class ScalingSource(str, Enum):
+    DEFAULT = "default"
+    CUSTOM = "custom"
+    EXTERNAL = "external"
+
+
+@PublicAPI(stability="alpha")
+class ScalingStatus(str, Enum):
+    SCALING_UP = "scaling_up"
+    SCALING_DOWN = "scaling_down"
+    STABLE = "stable"
+
+
+@PublicAPI(stability="alpha")
+class ScalingDecision(BaseModel):
+    """One autoscaling decision with minimal provenance."""
+
+    timestamp_s: float = Field(
+        ..., description="Unix time (seconds) when the decision was made."
+    )
+    source: ScalingSource = Field(
+        ..., description="Who made this decision: default/custom/external."
+    )
+    reason: str = Field(
+        ..., description="Short, human-readable reason for the decision."
+    )
+    from_replicas: int = Field(
+        ..., ge=0, description="Replica count before the decision."
+    )
+    to_replicas: int = Field(..., ge=0, description="Replica count after the decision.")
+    policy: Optional[str] = Field(
+        None, description="Policy name or identifier (if applicable)."
+    )
+    metrics: Optional[Dict[str, Any]] = Field(
+        None, description="Snapshot of metrics consulted when making the decision."
+    )
+
+
+@PublicAPI(stability="alpha")
+class DeploymentAutoscalerView(BaseModel):
+    """Deployment-level autoscaler observability."""
+
+    class Config:
+        extra = Extra.forbid
+
+    name: str = Field(..., description="Deployment name.")
+    current_replicas: int = Field(
+        ..., ge=0, description="Current number of live replicas."
+    )
+    proposed_replicas: int = Field(
+        ..., ge=0, description="Proposed number of replicas from the autoscaler policy."
+    )
+    min_replicas: Optional[int] = Field(
+        None, ge=0, description="Effective min replicas (capacity-adjusted)."
+    )
+    max_replicas: Optional[int] = Field(
+        None, ge=0, description="Effective max replicas (capacity-adjusted)."
+    )
```

Comment:
`name`, `current_replicas` and `proposed_replicas` is already available on `DeploymentDetails` object.
`min_replicas`, `max_replicas` and `lookback_period_s` is also available on `DeploymentDetails.deployment_config..`

what is the rationale for denormalizing the attributes?